### DISCUSSION TITLE
Pages looking funny after theme update

### DISCUSSION THREAD

**Discussion by HugoCebrian** [gregsullivan/_tw#82] [Q&A] [answered]
First of all, this is an amazing theme. I was able to build a nice website with it.

I'm constantly adding new stuff or changing how a specific section looks, and my only issue is when i have to push these updates. The page i changed will always look weird and need a full browser refresh to look normal like test environment. I know that it's cache-related, but i haven't found any solution that works. 

Anyone can give a light? How to update pages and force the cache to reload for users?

**Comment by gregsullivan** [comment]
Hello—Thanks for posting this, and I'm very happy to hear you like the theme!

Broadly speaking, browsers will cache static assets like CSS and JavaScript files, so you need to tell the browser that those files have updated if you want users (including yourself) who've visited the site already to load the new versions.

_tw uses the `_TW_VERSION` constant at the top of `./theme/functions.php` as the `$ver` argument in `wp_enqueue_style()` ([documentation](https://developer.wordpress.org/reference/functions/wp_enqueue_style/)).

If you generated your theme before April 13, 2023, or you don't use `npm run bundle` as part of your deployment process, you need to update this variable manually to force browsers to load new versions of theme assets.

If you generated your theme after April 13, 2023, running `npm run bundle` automatically uses date-based versioning, and you don't need to update the version number constant yourself.

If you're not sure whether you're getting automatic date-based versioning, please run `npm run bundle` and look for this as the final output from that command:

```
Date-based versioning string successfully added to `./functions.php`.
```

**Reply by HugoCebrian** [reply]
That's nice! I wasn't aware that _tw had a solution for that. 

My theme was generated last year, and i do use `npm run bundle`, but i have this issue since the start. 

Also, i don't see the output you mentioned in the console log. What should i do?

**Reply by gregsullivan** [reply, marked as answer]
While I don't officially document the upgrade process between different versions, I think this one would be as follows:

1. Run `npm install adm-zip --save-dev`
2. Replace the full contents of `./node_scripts/zip.js` in your repository with [this version](https://github.com/gregsullivan/_tw/blob/master/node_scripts/zip.js)

If you encounter any error messages when running `npm run bundle` after following the above steps, please let me know!

**Reply by HugoCebrian** [reply]
Hey! Sorry for the delay. I didn't have time to test until today. 

It seems like the building process is still working. I'll push a theme update today and will take notes on the results. 

Thanks a lot, Greg!

**Reply by gregsullivan** [reply]
No problem at all—very happy to hear it seems to be working!
